### PR TITLE
[CIRCSTORE-147, CIRCSTORE-148] Create CRUD API endpoints for patron action session

### PR DESCRIPTION
**Pull request by roman-barannyk**
## Purpose

Resolves: CIRCSTORE-147, CIRCSTORE-148

1. Create DB table to store check-in/check-out session information 
2. Create CRUD API endpoints for patron action session

## Approach
1. Update `AssertingRecordClient` with new field `String collectionPropertyName`
2. Add `attemptDeleteById(UUID id)` and `attemptPutById(JsonObject updateObject)` methods to `AssertingRecordClient` 

## TODOS and Open Questions
1. The name of the table and API endpoints are under discussion

## Links 
1. https://issues.folio.org/browse/CIRCSTORE-147
2. https://issues.folio.org/browse/CIRCSTORE-148

### CODE REVIEW

File: src/main/resources/templates/db_scripts/schema.json [hunk 1]
```diff
@@ -183,6 +183,11 @@
       "tableName": "scheduled_notice",
       "withMetadata": true,
       "withAuditing": false
+    },
+    {
+      "tableName": "patron_action_session",
```
[the remark this answers is not in the document]
Comment (dmytrotur) [reply src/main/resources/templates/db_scripts/schema.json]: @bohdan-suprun `patronId` is the reference to record from another module (`mod-users`).
About `loanId` i think it makes sense.

File: src/test/java/org/folio/rest/api/PatronActionSessionAPITest.java [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+  @Test
+  public void canCreatePatronActionSession() throws InterruptedException,
+    MalformedURLException, TimeoutException, ExecutionException {
+
+    JsonObject request = new JsonObject()
+      .put("id", UUID.randomUUID().toString())
+      .put("patronId", UUID.randomUUID().toString())
+      .put("loanId", UUID.randomUUID().toString())
+      .put("actionType", "Check-out");
+    assertRecordClient.create(request);
```
[the remark this answers is not in the document]
Comment (bohdan-suprun) [reply src/test/java/org/folio/rest/api/PatronActionSessionAPITest.java]: @dmytrotur, Yep, I'd suggest to add, just to improve test readability.

File: src/test/java/org/folio/rest/api/PatronActionSessionAPITest.java [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+    assertThat(patronIds, containsInAnyOrder(
+      firstSession.getString("patronId"),
+      secondSession.getString("patronId")));
+  }
+
+  @Test
+  public void canGetPatronActionSessionsByQueryAndLimit() throws InterruptedException, MalformedURLException,
+    TimeoutException, ExecutionException {
+
+    JsonObject firstSession = new JsonObject()
```
[the remark this answers is not in the document]
Comment (roman-barannyk) [reply src/test/java/org/folio/rest/api/PatronActionSessionAPITest.java]: @bohdan-suprun, thank you for review, tests have been updated

File: src/test/java/org/folio/rest/api/PatronActionSessionAPITest.java (hunk 2 above)
[the remark this answers is not in the document]
Comment (roman-barannyk) [reply src/test/java/org/folio/rest/api/PatronActionSessionAPITest.java]: @bohdan-suprun added

File: src/main/resources/templates/db_scripts/schema.json (hunk 1 above)
[the remark this answers is not in the document]
Comment (roman-barannyk) [reply src/main/resources/templates/db_scripts/schema.json]: @bohdan-suprun foreign key for loanId has been added

**Pull request merged**